api/src/shop/stripe_checkout.py

```diff
@@ -0,0 +1,252 @@
+        member.stripe_customer_id = stripe_customer['id']
+        # We set the subscription schedule id in the membership_subscription_column as this
+        # is also used to indicate pending subscriptions
+        if (subscription_type == SubscriptionTypes.MEMBERSHIP):
+          member.stripe_membership_subscription_id = subscription_schedule['id']
+        elif (subscription_type == SubscriptionTypes.LAB):
+          member.stripe_labaccess_subscription_id = subscription_schedule['id']
+        db_session.flush()
+
+        return success_url
+
+    except Exception as e:
+      return {'error': str(e)}
+
+# FIXME: Do we want business logic to prevent users from unsubscribing within X months?
+
+
+def cancel_subscription(member_id: int, subscription_type: SubscriptionTypes):
+    try:
+        member: Member = db_session.query(Member).get(member_id)
+        stripe_customer = get_stripe_customer(member)
+
+        subscription_id = None
+
+        if (subscription_type == SubscriptionTypes.MEMBERSHIP):
+          subscription_id = member.stripe_membership_subscription_id
+        elif (subscription_type == SubscriptionTypes.LAB):
+          subscription_id = member.stripe_labaccess_subscription_id
+
+        if (subscription_id == None):
+          return
+
+        # The subscription might be a scheduled one so we need to check the id prefix
+        # to determine which API to use.
+        if (subscription_id.startswith('sub_sched_')):
+          stripe.SubscriptionSchedule.release(subscription_id)
+          # No events from schedules so we need to remove the corresponding reference from the member here
+          if (subscription_type == SubscriptionTypes.MEMBERSHIP):
+            member.stripe_membership_subscription_id = None
+          elif (subscription_type == SubscriptionTypes.LAB):
+            member.stripe_labaccess_subscription_id = None
+
+        elif (subscription_id.startswith('sub_')):
+          stripe.Subscription.delete(sid=subscription_id)
+
+        member.stripe_customer_id = stripe_customer['id']
+        db_session.flush()
+
+        return {"reload": True}
+    except Exception as e:
+      return str(e)  # FIXME: We could use some structured error reporting
+
+def create_stripe_checkout_session(member_id:int, data:any = None, mode='payment'):
+  print('Creating stripe checkout session')
+  checkout_session = None
+
+  member: Member = db_session.query(Member).get(member_id)
+  stripe_customer = get_stripe_customer(member)
+  metadata = {
+      MSMetaKeys.USER_ID: member_id,
+      MSMetaKeys.MEMBER_NUMBER: member.member_number,
+  }
+
+  if (mode == 'setup'):
+    checkout_session = stripe.checkout.Session.create(
+      payment_method_types=['card'],
+      mode=mode,
+      metadata=metadata,
+      customer=stripe_customer['id'],
+      success_url=data["success_url"],
+      cancel_url=data["cancel_url"],
+    )
+
+  # TODO: Handle normal shop payments?
+
+    return checkout_session.url
```

Comment:
```suggestion
    return checkout_session.url
  else:
    return None
```
best to make all possible return values explicit